feat: initial implementation (#1)
ipshipyard/p2p-forge: README.md

- Addresses of the form `A::C:D` can be converted either into their expanded form or into a condensed form by replacing
the `:`s with `-`s, like `A--C-D`
- When there is a `:` as the first or last character it must be converted to a 0 to comply with [rfc1123](https://datatracker.ietf.org/doc/html/rfc1123#section-2)
, so `::B:C:D` would become `0--B-C-D` and `1::` would become `1--0`

Other address formats (e.g. the dual IPv6/IPv4 format) are not supported

### Submitting Challenge Records

To claim a domain name like `<peerID>.libp2p.direct` requires:
1. The private key corresponding to the given peerID
2. A publicly reachable libp2p endpoint with 
   - one of the following libp2p transport configurations:
     - QUIC-v1
     - TCP or WS or WSS, Yamux, TLS or Noise
     - WebTransport
     - Other transports are under consideration (e.g. HTTP), if they are of interest please file an issue
   - the [Identify protocol](https://github.com/libp2p/specs/tree/master/identify) (`/ipfs/id/1.0.0`)

To set an ACME challenge send an HTTP request to the server (for libp2p.direct this is registration.libp2p.direct)
```shell

> curl -X POST "https://registration.libp2p.direct/v1/<peerID>/_acme-challenge" \

Reviewer: The endpoint doesn't need the PeerID in it:

```suggestion
curl -X POST "https://registration.libp2p.direct/v1/_acme-challenge" \
```

-H "Authorization: libp2p-PeerID bearer=\"<base64-encoded-opaque-blob>\""
-H "Content-Type: application/json" \
-d '{
  "value": "your_acme_challenge_token",

>   "addresses": "[your_multiaddrs, comma_separated]"

Reviewer: Is `addresses` a string that contains an string array in JSON notation, eg:

```json
{
  "value": "...",
  "addresses": "[\"/ip4/...\", \"/ip6/...\"]"
}
```

or is it a single string, eg:

```json
{
  "value": "...",
  "addresses": "/ip4/...,/ip6/..."
}
```

If so, why not just a simple array of strings, eg:

```json
{
  "value": "...",
  "addresses": ["/ip4/...", "/ip6/..."]
}
```

Reviewer: Looking at [the code](https://github.com/ipshipyard/p2p-forge/pull/1/files#diff-1db9aa3c1835054d2178f88673e32e5f18ed6b3a8e787df6865b89fe3e9b90eeR51) I think it is an array?

The example might be better as:

```suggestion
  "addresses": ["your", "multiaddrs"]
```

}'
```

> Where the bearer token is derived via the [libp2p HTTP PeerID Auth Specification](https://github.com/libp2p/specs/pull/564).

Reviewer: The spec can be linked to directly:

```suggestion
Where the bearer token is derived via the [libp2p HTTP PeerID Auth Specification](https://github.com/libp2p/specs/blob/master/http/README.md).
```
